npm/debugTools/dataplane/parser.go

```diff
@@ -0,0 +1,338 @@
+package dataplane
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+// Parser struct
+type Parser struct{}
+
+// ParseIptablesObject create a Go object from specified iptable by calling iptables-save within node
+func (p *Parser) ParseIptablesObject(tableName string) *Iptables {
+	iptableBuffer := bytes.NewBuffer(nil)
+	// TODO: need to get iptable's lock
+	cmdArgs := []string{"-t", string(tableName)}
+	cmd := exec.Command(util.IptablesSave, cmdArgs...) //nolint:gosec
+
+	cmd.Stdout = iptableBuffer
+	stderrBuffer := bytes.NewBuffer(nil)
+	cmd.Stderr = stderrBuffer
+
+	err := cmd.Run()
+	if err != nil {
+		_, err = stderrBuffer.WriteTo(iptableBuffer)
+		if err != nil {
+			panic(err)
+		}
+	}
+	chains := p.parseIptablesChainObject(tableName, iptableBuffer.Bytes())
+	return &Iptables{Name: tableName, Chains: chains}
+}
+
+// ParseIptablesObjectFile create a Go object from specified iptable by reading from an iptables-save file
+func (p *Parser) ParseIptablesObjectFile(tableName string, iptableSaveFile string) *Iptables {
+	iptableBuffer := bytes.NewBuffer(nil)
+	byteArray, err := ioutil.ReadFile(iptableSaveFile)
+	if err != nil {
+		fmt.Print(err)
+	}
+	for _, b := range byteArray {
+		iptableBuffer.WriteByte(b)
+	}
+	chains := p.parseIptablesChainObject(tableName, iptableBuffer.Bytes())
+	return &Iptables{Name: tableName, Chains: chains}
+}
+
+// parseIptablesChainObject create a map of iptable chain name and iptable chain object
+func (p *Parser) parseIptablesChainObject(tableName string, iptableBuffer []byte) map[string]*IptablesChain {
```

Comment:
Can you document the limitations we have with this parsing logic, or document what it can do today.